packages/core/src/account/smartContractAccount.ts

```diff
@@ -115,7 +116,7 @@ export type ToSmartContractAccountParams<
   Name extends string = string,
   TTransport extends Transport = Transport,
   TChain extends Chain = Chain,
-  TEntryPointVersion extends EntryPointVersion = EntryPointVersion
+  TEntryPointVersion extends EntryPointVersion = DefaultEntryPointVersion
```

Comment:
I think we need to leave the base definition as it was here

Reply:
typescript fails. because function overloaded without typing defaults to default entrypoint version, and we call toSmartContractAccount in that impl. so it thinks the entrypoint is 0.6.0 (default). This is one caveat of the current approach.

Reply:
@moldy530

Reply:
because there was a default EP being passed in yea, but if we don't pass in a default and make EP required then it's fine